doc/dox_comments/header_files/aes.h

```diff
@@ -658,6 +658,82 @@ int  wc_AesCcmDecrypt(Aes* aes, byte* out,
                                    const byte* authTag, word32 authTagSz,
                                    const byte* authIn, word32 authInSz);
 
+/*!
+    \ingroup AES
+
+    \brief This is to initialize an AES-XTS context. It is up to user to call
+    wc_AesXtsFree on aes key when done.
+
+    \return 0 Success
+
+    \param aes   AES keys for encrypt/decrypt process
+    \param heap  heap hint to use for memory. Can be NULL
+    \param devId id to use with async crypto. Can be 0
+
+    _Example_
+    \code
+    XtsAes aes;
+
+    if(wc_AesXtsInit(&aes, NULL, 0) != 0)
+    {
+        // Handle error
+    }
+    if(wc_AesXtsSetKeyNoInit(&aes, key, sizeof(key), AES_ENCRYPTION) != 0)
+    {
+        // Handle error
+    }
+    wc_AesXtsFree(&aes);
+    \endcode
+
+    \sa wc_AesXtsSetKey
+    \sa wc_AesXtsSetKeyNoInit
+    \sa wc_AesXtsEncrypt
+    \sa wc_AesXtsDecrypt
+    \sa wc_AesXtsFree
+*/
+int wc_AesXtsInit(XtsAes* aes, void* heap, int devId);
+
+
+/*!
+    \ingroup AES
+
+    \brief This is to help with setting keys to correct encrypt or decrypt type,
+    after first calling wc_AesXtsInit(). It is up to user to call wc_AesXtsFree
+    on aes key when done.
+
+    \return 0 Success
+
+    \param aes   AES keys for encrypt/decrypt process
+    \param key   buffer holding aes key | tweak key
+    \param len   length of key buffer in bytes. Should be twice that of
+    key size.
+                 i.e. 32 for a 16 byte key.
+    \param dir   direction, either AES_ENCRYPTION or AES_DECRYPTION
+
+    _Example_
+    \code
+    XtsAes aes;
+
+    if(wc_AesXtsInit(&aes, NULL, 0) != 0)
+    {
+        // Handle error
+    }
+    if(wc_AesXtsSetKeyNoInit(&aes, key, sizeof(key), AES_ENCRYPTION, NULL, 0)
+       != 0)
+    {
+        // Handle error
+    }
+    wc_AesXtsFree(&aes);
+    \endcode
+
+    \sa wc_AesXtsEncrypt
+    \sa wc_AesXtsDecrypt
+    \sa wc_AesXtsFree
+*/
+int wc_AesXtsSetKeyNoInit(XtsAes* aes, const byte* key,
```

Comment:
Is `wc_AesXtsSetKeyNoInit` the first and only `NoInit` type Aes API? (I think it is.)

It might be helpful to have an additional comment clarifying the difference between `wc_AesXtsSetKey()` and `wc_AesXtsSetKeyNoInit()`, since this is a special API.

Reply:
It's noted in the documentation added by this PR.

The awkwardness of the new API follows from an early oversight in which `wc_AesXtsSetKey()` has fundamentally different semantics from `wc_AesSetKey()`.  The latter requires that `wc_AesInit()` be called first -- meeting that requirement was the whole purpose of your #7011 -- whereas the former forbids it.  Unfortunately `wc_AesXtsSetKey()` is a public API, so its semantics can't be altered to be consistent with `wc_AesSetKey()`.